### PR TITLE
fluxible-addons-react: improve packaging

### PR DESCRIPTION
**Pull request by pablopalacios**
I did small packaging improvements to decrease final package size:
- Add `es` and `commonjs` builds to the final package (it should be tree-shakeable now);
- Remove `src` from final package;
- Use `babel-runtime` to decrease final modules size by not inlining babel helpers on each file;
- Convert `FluxibleComponent` to function to decrease final module size (4.2kb -> 0.7kb);

**Before**
```bash
npm notice === Tarball Contents === 
npm notice 1.1kB dist/batchedUpdatePlugin.js     
npm notice 766B  src/batchedUpdatePlugin.js      
npm notice 8.0kB dist/connectToStores.js         
npm notice 3.1kB src/connectToStores.js          
npm notice 2.4kB dist/createElementWithContext.js
npm notice 1.1kB src/createElementWithContext.js 
npm notice 4.2kB dist/FluxibleComponent.js       
npm notice 700B  src/FluxibleComponent.js        
npm notice 4.6kB dist/FluxibleContext.js         
npm notice 1.2kB src/FluxibleContext.js          
npm notice 1.5kB dist/index.js                   
npm notice 534B  src/index.js                    
npm notice 6.4kB dist/provideContext.js          
npm notice 2.0kB src/provideContext.js           
npm notice 937B  package.json                    
npm notice 2.5kB CHANGELOG.md                    
npm notice 338B  CONTRIBUTING.md                 
npm notice 1.7kB LICENSE.md                      
npm notice 1.6kB README.md                       
npm notice 2.7kB UPGRADE.md                      
npm notice === Tarball Details === 
npm notice name:          fluxible-addons-react                   
npm notice version:       1.0.0-beta.0                            
npm notice package size:  9.5 kB                                  
npm notice unpacked size: 47.3 kB                                 
npm notice shasum:        6a89b0a24e31f1f6fcf443edf28409b53b85610e
npm notice integrity:     sha512-z1Oludsi8OQI0[...]Nmb7hat97SjUw==
npm notice total files:   20      
```

**After**
```bash
npm notice === Tarball Contents === 
npm notice 914B  dist/es/batchedUpdatePlugin.js      
npm notice 1.1kB dist/lib/batchedUpdatePlugin.js     
npm notice 4.3kB dist/es/connectToStores.js          
npm notice 4.9kB dist/lib/connectToStores.js         
npm notice 1.2kB dist/es/createElementWithContext.js 
npm notice 1.5kB dist/lib/createElementWithContext.js
npm notice 732B  dist/es/FluxibleComponent.js        
npm notice 912B  dist/lib/FluxibleComponent.js       
npm notice 1.7kB dist/es/FluxibleContext.js          
npm notice 2.2kB dist/lib/FluxibleContext.js         
npm notice 533B  dist/es/index.js                    
npm notice 1.5kB dist/lib/index.js                   
npm notice 2.6kB dist/es/provideContext.js           
npm notice 3.1kB dist/lib/provideContext.js          
npm notice 1.2kB package.json                        
npm notice 2.5kB CHANGELOG.md                        
npm notice 1.7kB LICENSE.md                          
npm notice 1.6kB README.md                           
npm notice === Tarball Details === 
npm notice name:          fluxible-addons-react                   
npm notice version:       1.0.0-beta.0                            
npm notice package size:  6.9 kB                                  
npm notice unpacked size: 34.1 kB                                 
npm notice shasum:        1972cbbb0bd1cacd05b641524b4d91ff19580786
npm notice integrity:     sha512-X+kff6dcBpPMU[...]DL4eheAmtT60g==
npm notice total files:   18           
```

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/fluxible-addons-react/.babelrc.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+    "plugins": [
+        [
+            "@babel/plugin-transform-runtime",
+            {
+                "version": "7.14.0"
```
Comment (redonkulus) [inline packages/fluxible-addons-react/.babelrc.json:6]: Why is version specified?

**Review by redonkulus — COMMENTED** [review]
Looks good overall, nice to see the bundle size improvements. Just a couple of comments.

**Comment by pablopalacios** [comment]
@redonkulus I'm replying here since both questions are related.

First of all, a little bit of context:

When babel transpiles the code, it might include some helper functions in the output file. As an example: 

```javascript
class Circle {}
```

Would become:

```javascript
function _classCallCheck(instance, Constructor) {
  //...
}

var Circle = function Circle() {
  _classCallCheck(this, Circle);
};
```
In the example above, `_classCallCheck` is a babel helper function that would be included and duplicated in every file that has a class defined.

By using `@babel/plugin-transform-runtime` transform plugin, instead of duplicating  `_classCallCheck` definition, babel will import this function from `@babel/runtime` package. The above example would become:

```javascript
var _classCallCheck = require("@babel/runtime/helpers/classCallCheck");

var Circle = function Circle() {
  _classCallCheck(this, Circle);
};
```
As you can see, we are importing a function from `@babel/runtime` and for this reason, we must add it as a real dependency (instead of a dev one).

Regarding the specified version in `.babelrc.json`, if we don't do it, babel will default to version `7.0.0` (as documented [here](https://babel.dev/docs/en/babel-plugin-transform-runtime#version)). It's not a big problem to keep the default version in our case, since the package would still work. However, there are some helper functions that were not being deduplicated since support for them arrived only in later versions of `@babel/runtime`.

Enough of theory, here are the 3 versions of `connectToStores.js` module so you can take a look at the output.

1. Without @babel/runtime (7.9K): https://termbin.com/z2dm
2. Only @babel/runtime (6.4K): https://termbin.com/ormr
3. With @babel/runtime and version set (4.8K): https://termbin.com/b59q

By the way, I more or less copied all these babel features from `react-redux` package. You can check their configuration [here](https://github.com/reduxjs/react-redux/blob/master/.babelrc.js#L10). As you can see, they derive the version from the package.json which is quite cool. But given that we don't update `fluxible-addons-react` quite often, I though that it would be a bit too much to do what they did there.

File: packages/fluxible-addons-react/package.json [hunk 2]
```diff
@@ -17,6 +20,7 @@
     "lint": "../../node_modules/.bin/eslint src/*.js tests/"
   },
   "dependencies": {
+    "@babel/runtime": "^7.14.0",
```
[the remark this answers is not in the document]
Comment (pablopalacios) [reply packages/fluxible-addons-react/package.json:23]: Replied here: https://github.com/yahoo/fluxible/pull/683#issuecomment-830857300

File: packages/fluxible-addons-react/.babelrc.json (hunk 1 above)
Comment (pablopalacios) [reply packages/fluxible-addons-react/.babelrc.json:6]: Replied here: https://github.com/yahoo/fluxible/pull/683#issuecomment-830857300

**Comment by pablopalacios** [comment]
As a note why I prepared this PR, the master version was quite small and I completely screwed it:

https://bundlephobia.com/result?p=fluxible-addons-react@1.0.0-beta.0

```bash
npm notice === Tarball Contents === 
npm notice 1.2kB batchedUpdatePlugin.js     
npm notice 4.5kB connectToStores.js         
npm notice 1.1kB createElementWithContext.js
npm notice 1.2kB FluxibleComponent.js       
npm notice 6.6kB FluxibleMixin.js           
npm notice 499B  index.js                   
npm notice 2.9kB provideContext.js          
npm notice 823B  package.json               
npm notice 2.5kB CHANGELOG.md               
npm notice 338B  CONTRIBUTING.md            
npm notice 1.7kB LICENSE.md                 
npm notice 1.8kB README.md                  
npm notice === Tarball Details === 
npm notice name:          fluxible-addons-react                   
npm notice version:       0.2.16                                  
npm notice package size:  7.0 kB                                  
npm notice unpacked size: 25.1 kB                                 
npm notice shasum:        7c9ed8e60380723f72e1f3404cdb8aef3f2f12dc
npm notice integrity:     sha512-r//EukN1jamFv[...]W8Od0yOxk/2/A==
npm notice total files:   12                                      
```

But with the `es` version that I prepared here (which would be the one taken by webpack), is now smaller and tree-shakeable:

```bash
npm notice === Tarball Contents === 
npm notice 914B  dist/es/batchedUpdatePlugin.js     
npm notice 4.3kB dist/es/connectToStores.js         
npm notice 1.2kB dist/es/createElementWithContext.js
npm notice 732B  dist/es/FluxibleComponent.js       
npm notice 1.7kB dist/es/FluxibleContext.js         
npm notice 533B  dist/es/index.js                   
npm notice 2.6kB dist/es/provideContext.js          
npm notice 1.2kB package.json                       
npm notice 2.5kB CHANGELOG.md                       
npm notice 1.7kB LICENSE.md                         
npm notice 1.6kB README.md                          
npm notice === Tarball Details === 
npm notice name:          fluxible-addons-react                   
npm notice version:       1.0.0-beta.0                            
npm notice package size:  5.8 kB                                  
npm notice unpacked size: 18.8 kB                                 
npm notice shasum:        28e4ecf1a09f49355c9dd57a24ce22a153215619
npm notice integrity:     sha512-iJejT8gHQ3piG[...]gCg+F7PADmxLw==
npm notice total files:   11                                      
```

**Comment by redonkulus** [comment]
Ok great explanation, makes to me!

**Pull request merged**